pateikti failai

Edgaras318 opened the pull request.

Commit:
pateikti failai

Commit:
Refactor Register page and add reusable components

Replaced existing form logic in the Register page with a new RegisterForm component. Created a reusable InputField component and moved validation logic to a separate validators file for better code organization and maintainability.

Commit:
Refactor InputField and Login page components

Updated InputField for better modularity and usability. Refactored the Login page to utilize reusable InputField and relocated validation logic to a separate validators file for improved code organization.

Commit:
Refactor component names and paths for consistency

Renamed various components and their file paths to improve clarity and maintain consistent naming conventions across the project. Adjusted import statements and JSX references accordingly to reflect these changes.

Commit:
Remove unnecessary blank line in BusinessCardList

Deleted an extra blank line in the BusinessCardList component to improve code readability and maintain a clean coding style. This change does not affect the functionality of the component.

Commit:
Update category component imports and centralize category constants

Standardized imports across components by renaming and correcting paths for VerticalCategoriesSection. Centralized category data into a single file for easier maintenance and consistency.

ZilvinasVidmantas commented on .idea/modules.xml:
.idea failus reiktų už-ignore-uoti .gitignore

Edgaras318 replied:
18 eilutej uzignorinti?

ZilvinasVidmantas commented on src/App.jsx:
```diff
@@ -0,0 +1,47 @@
+
+const router = createBrowserRouter([
+  {
+    path: "/",
```
Route'ų nuorodoms(url adresams) geriau suskurti atskirą objektą, ir naudoti jį puslapio navigacijose ir Route'ų subuildinime

ZilvinasVidmantas commented on src/components/BusinessCard/BusinessCard.jsx:
```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import Button from "../common/Button/Button.jsx";
```
Geriau naudoti globalius importus

ZilvinasVidmantas commented on src/components/BusinessCard/BusinessCard.jsx:
```diff
@@ -0,0 +1,45 @@
+import styles from "./BusinessCard.module.scss";
+
+const BusinessCard = ({ business }) => {
+    const [favorites, setFavorites] = useLocalStorage('favorites', []);
```
Kiekviena kortelė parsitraukia tą patį localSorage kintamajį. Perteklinis naudojimas.
Geriau būtų Kortelės tėvineme komponente laikyti funkciją, gauti duomenis ir keisti localStorage vertę, o visus kortelei reikalingus duomenis perduoti per prop'sus

ZilvinasVidmantas commented on src/components/CategoryCard/CategoryCard.module.scss:
```diff
@@ -0,0 +1,34 @@
+  align-items: center;
+  justify-content: center;
+  gap: 0.8rem;
+  background-color: #f2eefd;
```
reikėtų naudoti sccs/css kintamuosius, kad vbėliau lengviau keisti temą:

ZilvinasVidmantas commented on src/components/RegisterForm/RegisterForm.jsx:
```diff
@@ -0,0 +1,91 @@
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setForm((prev) => ({ ...prev, [name]: value }));
+        setErrors((prev) => ({ ...prev, [name]: "" }));
```
Jei klaidos prieš tai nebuvo, spreadinant susikurs nauja objekto nuorodą ir pertekliškai persikraus komponentas. Geriau būtų įdėti if'ą - ir klaidas perrašyti tik tuomet jeigu klaida egzistuoja [name] raktu

ZilvinasVidmantas commented on src/components/Topbar/Topbar.jsx:
```diff
@@ -0,0 +1,89 @@
+    }
+  }, [user]);
+
+  const renderUserAvatar = () => {
```
Čia yra komponentas, kurti komponente komponentą nėra gera praktika. Reiktų iškelti į atskirą failą, o priklausomybes aprašyti per propsus, ir juos perduoti

ZilvinasVidmantas replied:
https://stackoverflow.com/questions/71658569/react-whats-the-difference-between-a-function-returning-jsx-and-function-compo

Commit:
git commit -m "Update .gitignore to exclude .idea directory"

Commit:
Add centralized routes configuration file

Created a new routes.js file to centralize route paths and updated all relevant components to use this new configuration. This improves maintainability by having all route paths in a single location. Fixed a typo in the About Us component.

Commit:
Refactor BusinessCard component imports

Updated the import path for the Button component to use an absolute path. This change aims to improve code readability and maintainability by standardizing import paths across the project. Removed an unnecessary useState import.

Commit:
Simplify Topbar and abstract user avatar

Moved user avatar logic and styles to a new UserAvatar component for better separation of concerns. This refactor improves code readability and maintainability by isolating specific functionality into distinct components. Updated Topbar to utilize the new UserAvatar component.

Commit:
Add favorites functionality to BusinessCardList

Implemented state management for favorites using local storage in BusinessCardList. Updated BusinessCard component to accept favorite status and toggle function as props, ensuring better separation of concerns and improved code maintainability.

Commit:
Refactor card styling using CSS variables

Centralized card colors in _variables.scss and updated CategoryCard and VerticalCategoryCard styles to use these variables, enhancing maintainability and consistency across the project. This change simplifies theme adjustments and reduces the likelihood of color mismatches.

Commit:
Improve error state update logic in RegisterForm

Modified handleChange to conditionally clear existing error messages for form fields. This ensures the state is only updated when necessary, enhancing performance and maintaining clarity in error handling.

Commit:
Refactor form handling in Login component

Unified username and password into a single state object and added a clear error logic when the corresponding field changes. Also introduced name attributes to InputField components for better state management.
